src/main/blackbox/blackbox.c

```diff
@@ -1657,7 +1657,8 @@ void handleBlackbox(timeUs_t currentTimeUs)
 
 static bool canUseBlackboxWithCurrentConfiguration(void)
 {
-    return feature(FEATURE_BLACKBOX);
+    return feature(FEATURE_BLACKBOX) && 
+        (blackboxConfig()->device != BLACKBOX_SDCARD || feature(FEATURE_SDCARD));
```

Comment:
`!(blackboxConfig()->device == BLACKBOX_SDCARD && !feature(FEATURE_SDCARD))` makes it more obvious what this is guarding against.